src/lua/fiber.c

```diff
@@ -810,6 +810,113 @@ lbox_fiber_stall(struct lua_State *L)
 	return 0;
 }
 
+/* Increase deadline timeout of currently running fiber. */
+static int
+lbox_extend_slice(struct lua_State *L)
+{
+	if (lua_gettop(L) != 2) {
+		luaL_error(L, "fiber.extend_slice(fib_id, slice): "
+			      "bad arguments");
+	}
+	struct fiber *fiber = lbox_checkfiber(L, 1);
+	if (fiber != fiber()) {
+		luaL_error(L, "fiber is not running");
```

Comment:
Why do you need the argument if you require that this function must only be called on the current fiber? Let's drop this argument and move this function to the fiber module, like you did with `fiber_check_slice`.

Reply:
@unera asked to move this function to *fiber object* not to clutter up `fiber` module.

Reply:
Since you wrote below that *unera* agreed not to pass fiber as argument to `extend_slice`, I will fix it.

Reply:
fixed